source/Quantities.Benchmark/Exporter.cs

```diff
@@ -11,32 +11,40 @@ public static class Exporter
 
     public static void Export(IEnumerable<Summary> summaries)
     {
-        var sourceDir = FindSourceDir();
+        var sourceFiles = FindSourceFiles();
         foreach (var summary in summaries) {
-            Export(summary, sourceDir);
+            Export(summary, sourceFiles);
         }
     }
-    public static void Export(Summary summary) => Export(summary, FindSourceDir());
-    private static void Export(Summary summary, DirectoryInfo sourceDir)
-        => Export(summary, MarkdownExporter.Console, sourceDir);
-    private static void Export(Summary summary, IExporter exporter, DirectoryInfo sourceDir)
+    public static void Export(Summary summary) => Export(summary, FindSourceFiles());
+    private static void Export(Summary summary, List<FileInfo> allFiles)
+        => Export(summary, MarkdownExporter.Console, allFiles);
+    private static void Export(Summary summary, IExporter exporter, List<FileInfo> allFiles)
     {
         foreach (var file in exporter.ExportToFiles(summary, logger)) {
             var name = BenchmarkName(file);
             var fileName = $"{name}.cs";
             Console.WriteLine($"Exporting: {name}");
-            var sourceFile = sourceDir.EnumerateFiles("*.cs", SearchOption.AllDirectories).Single(f => f.Name.EndsWith(fileName));
+            var sourceFile = allFiles.Single(f => f.Name.EndsWith(fileName));
             UpdateSourceFile(sourceFile, File.ReadAllText(file));
             File.Delete(file);
         }
     }
     private static void UpdateSourceFile(FileInfo source, String result)
     {
         var tmpFile = $"{source.FullName}.tmp";
-        using var reader = source.OpenText();
-        using var writer = File.CreateText(tmpFile);
         try {
+            Update(source, tmpFile, result);
+        }
+        finally {
+            File.Move(tmpFile, source.FullName, overwrite: true);
```

Comment:
what if the tmpFile could not be created because of permissions or some other weird reasons?